### DISCUSSION TITLE
Suggestion to Add Shared Server Play Count

### DISCUSSION THREAD

**Discussion by thebiblelover7** [opensubsonic/open-subsonic-api#96] [General]
I think it would be pretty nice to have a shared play count for a track. It would be counted across multiple users.
It would be pretty cool in our server's case so that "popular" songs will work a bit more like Spotify. That way, recommendations done by Feishin or other clients may be more useful.

<img width="871" alt="Screenshot 2024-07-02 at 4 23 14 PM" src="https://github.com/navidrome/navidrome/assets/61815862/56194076-b6e7-4704-898b-0965622f8f96">

Related: navidrome/navidrome#3131

**Comment by spl0k** [comment]
I guess that calls for a clarification on that field.

If I understand correctly, the play count in Navidrome is tied to the current user. In Supysonic (which isn't OpenSubsonic compatible but that's just for the example) the play count is shared across all users. As the description for that field is vague I'm expecting different servers might have different implementations.

**Reply by Tolriq** [reply]
Well if there's already differences on the current field on some major servers, it's not really possible to clarify the field.

We need to add 2 new fields with proper meaning in that case.

**Comment by noah1234j** [comment]
+1 for this. Maybe its a flag that can be flipped in the config? Rather than adding a separate API field. Either way I would love to see this implemented

**Reply by thebiblelover7** [reply]
That would be really nice!

**Reply by lachlan-00** [reply]
I always assumed most servers used the global count but if not, I wouldn't change this field using a preference or config.

Either all servers use a global count and add userCount or you add 2 new fields as above and ignore the old one.

I can't find a definitive answer on what the count fields should be so it's better left alone.

**Reply by noah1234j** [reply]
I see where you're coming from. If user-specific play count is what all the players and clients expect would probably mess some things up to allow people to flip it in the config. Sounds like new fields might be the best direction to go here. I think there's a number of users that would use it.

**Reply by noah1234j** [reply]
I'll do a bit of digging in the next few weeks when I have time and see if I can maybe help get this contributed. I guess custom fields would only be useful to those who are developing their own custom player tho. People who are just wanting to see global play counts in their existing apps would want to be able to flip in in the config. Either is fine by me as I am using my own custom player.

**Reply by noah1234j** [reply]
@lachlan-00 I guess the the quesion is if I were to build out the config option where users could flip it in the server config. Would it make it to the main branch? I wouldn't really be interested in doing it if its just going to become another fork. Either way I understand.

**Reply by lachlan-00** [reply]
I see in the subsonic spec playCount is optional.
```
<xs:attribute name="playCount" type="xs:long" use="optional"/>
```

I would stick with making 2 new attributes and the server can double up the old one
```
<xs:attribute name="count" type="xs:long" use="optional"/>
<xs:attribute name="userCount" type="xs:long" use="optional"/>
<xs:attribute name="playCount" type="xs:long" use="optional"/>
```

that way all the servers can provide something clear and optionally keep the old one for compatibility. then just note that playCount should be ignored/deprecated for removal later by open subsonic servers

**Reply by lachlan-00** [reply]
pull to do that here https://github.com/opensubsonic/open-subsonic-api/pull/144

**Reply by noah1234j** [reply]
Ok will work on it when possible thank you!
